cadence/contracts/bridge/BridgePermissions.cdc

```diff
@@ -0,0 +1,17 @@
+/// This contract defines a simple interface which can be implemented by any resource to prevent it from being
+/// onboarded to the Flow-EVM bridge
+///
+/// NOTE: This is suggested only for cases where your asset (NFT/FT) incorporates non-standard logic that would
+///      break your project if not handles properly
+///      e.g. assets are reclaimed after a certain period of time, NFTs share IDs, etc.
+///
+access(all)
+contract interface BridgePermissions {
+    /// Contract-level method enabling implementing contracts to identify whether they allow bridging for their
+    /// project's assets. Implementers may consider adding a hook which would later enable an update to this value
+    /// should either the project be updated or the bridge be updated to handle the asset's non-standard logic which 
+    /// would otherwise prevent them from supporting VM bridging at the outset.
+    ///
+    access(all)
+    view fun allowsBridging(): Bool
```

Comment:
The idea here is that bridging is enabled for all contracts on either side by default. There are however some non-standard use cases (which would be a small minority) as @dete pointed out that might arise where bridging would fundamentally break product expectations - e.g. an NFT can be reclaimed after so much inactivity. In such cases, it would make sense for projects to opt-out of bridging because bridge assumptions (based on existing NFT & FT standards of ownership) would compromise the project's business logic.

> Can this be handled as part of the new bridge metadata view? It seems like it's bridge specific config being set by the project owner.

I'm sure it could be, but because a/ there's an expectation that the implementing contracts would be a minority of cases and b/ it would be helpful for opt-out mechanisms between VMs to leverage consistent mental models (i.e. contract interfaces), IMO it makes more sense for it to be a very intentional implementation detail that is clearly non-standard.

> Will there be a per-NFT type of access control needed in the future? Imagine a project that has NFTs in two states: processing and final and they want to only allow bridging for final NFTs.

For the cases discussed where the need for an opt-out mechanism was raised, I don't think we'd want to introduce the added complexity in both implementation and user expectations required to support per-NFT permissions. I could see it getting very confusing for users and user-serving products if they need to assess bridge permissions on a per-token basis.

But curious if you had some specific use cases in mind. Does that all sound reasonable?

Reply:
Yes it makes sense. Thank you for the explanation. I think we can leave any future developments like per-NFT bridging permissions to the future.